src/misc/dbg_helpers.cpp

```diff
@@ -56,20 +54,19 @@ static const char * const signal_type_names[] = {
 };
 
 /** Return name of given SignalType. */
-CStrA ValueStr(SignalType t)
+std::string ValueStr(SignalType t)
 {
-	CStrA out;
-	out.Format("%d (%s)", t, ItemAtT(t, signal_type_names, "UNK"));
-	return out.Transfer();
+	return std::to_string(t) + " (" + ItemAtT(t, signal_type_names, "UNK") + ")";
 }
 
 
 /** Translate TileIndex into string. */
-CStrA TileStr(TileIndex tile)
+std::string TileStr(TileIndex tile)
 {
-	CStrA out;
-	out.Format("0x%04X (%d, %d)", tile, TileX(tile), TileY(tile));
-	return out.Transfer();
+	std::stringstream ss;
```

Comment:
personally I always do "char buf[4]" and "seprintf" in such cases.
But now you already put in the effort to find those weird stream modifiers :p

Reply:
Depends how much we care about the extra overhead of sstream/iomanip includes